workspace/solution.go

```diff
@@ -0,0 +1,46 @@
+package workspace
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const solutionFilename = ".solution.json"
+
+// Solution contains metadata about a user's solution.
+type Solution struct {
+	Track       string    `json:"track"`
+	Exercise    string    `json:"exercise"`
+	ID          string    `json:"id"`
+	URL         string    `json:"url"`
+	Handle      string    `json:"handle"`
+	IsRequester bool      `json:"is_requester"`
+	SubmittedAt time.Time `json:"submitted_at"`
```

Comment:
I just realized that we will sometimes download a fresh exercise where the solution hasn't been submitted yet. There will still be a _solution_. It has an ID, URL, etc. It just hasn't been submitted, so there's no timestamp.

I think I need a custom time format that can be a pointer.